assets/components/directDebit/directDebitActions.js

```diff
@@ -67,13 +67,20 @@ function payDirectDebitClicked(callback: Function): Function {
     }
 
     checkAccount(bankSortCode, bankAccountNumber, isTestUser, csrf)
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error('code1');
+        }
+        return response.json();
+      })
       .then((response) => {
         if (!response.accountValid) {
           throw new Error('code1');
         }
       }).then(() => {
         callback(undefined, bankAccountNumber, bankSortCode, accountHolderName);
-      }).catch((e) => {
+      })
+      .catch((e) => {
         let msg = '';
         switch (e.message) {
           case 'code1': msg = 'Your payment details are invalid. Please check them and try again';
```

Comment:
Should the default message be something like 'Oops, something went wrong, please try again later'? Now the catch function will pick up things like our JSON response being malformed.

Reply:
Yeap, I will change the default case. Nevertheless, this is only catching part of the entire cases.

Reply:
Updated the default case here: https://github.com/guardian/support-frontend/pull/460/commits/f8f96c12e0435205a553e4193b97d9e2d4ca8070